# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/events/EventStateManager.cpp
# Commit: e974d872becd
# Full Hash: e974d872becd4fae0214738cdf27173a994d0f59
# Author: Dan Robertson <drobertson@mozilla.com>
# Date: 2023-08-30 21:27:31
# Regressor Bug: 1816473
# File Overlap Count: 3
# Description:
#   Bug 1816473 - Do not delay dispatching synthesized mouse events. r=botond,edgar
#   
#   Do not delay dispatching synthesized mouse events for a single tap
#   gesture.
#   
# ==============================================================================

diff -r 1b272e14e402 -r e974d872becd dom/events/EventStateManager.cpp
--- a/dom/events/EventStateManager.cpp	Wed Aug 30 12:06:41 2023 +0000
+++ b/dom/events/EventStateManager.cpp	Wed Aug 30 12:49:22 2023 +0000
@@ -3642,8 +3642,13 @@
       // we should never be capturing when the mouse button is up
       PresShell::ReleaseCapturingContent();
 
-      ClearGlobalActiveContent(this);
       WidgetMouseEvent* mouseUpEvent = aEvent->AsMouseEvent();
+      // If the mouseup event is a synthesized mouse event due to a touch, do
+      // not clear the activation state. Element activation is handled by APZ.
+      if (!mouseUpEvent || mouseUpEvent->mInputSource !=
+                               dom::MouseEvent_Binding::MOZ_SOURCE_TOUCH) {
+        ClearGlobalActiveContent(this);
+      }
       if (mouseUpEvent && EventCausesClickEvents(*mouseUpEvent)) {
         // Make sure to dispatch the click even if there is no frame for
         // the current target element. This is required for Web compatibility.